cli/wallet/wallet.go

```diff
@@ -542,27 +557,61 @@ func importMultisig(ctx *cli.Context) error {
 		}
 	}
 
-	var label *string
 	if ctx.IsSet("name") {
 		l := ctx.String("name")
 		label = &l
 	}
-	acc, err := newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
-	if err != nil {
-		return cli.NewExitError(err, 1)
-	}
 
-	if err := acc.ConvertMultisig(m, pubs); err != nil {
-		return cli.NewExitError(err, 1)
+	for _, pub := range pubs {
+		match := findAccountsByPublicKey(wall, pub)
+		if len(match) > 0 {
```

Comment:
Do not add them to the list. Remove `matchingAccounts`, it can be handled without additional vaiable. Check if len(match) >0 and assign `acc` to `match[0]`. After that check `if len(match) > 1` and fallback to WIF if true. Continue traversal and if acc is already set and there are more `match` then fallback to WIF.